server/src/main/java/org/apache/iotdb/db/mpp/execution/memory/MemoryPool.java

```diff
@@ -154,6 +154,14 @@ public Pair<ListenableFuture<Void>, Boolean> reserve(
         bytesToReserve > 0L && bytesToReserve <= maxBytesPerFragmentInstance,
         "bytes should be greater than zero while less than or equal to max bytes per fragment instance: %d",
         bytesToReserve);
+    if (bytesToReserve > maxBytesCanReserve) {
+      LOGGER.error(
```

Comment:
warn level log

Reply:
Why? It's an error which means it will never make it that reserve enough memory, and supposed to throw an exception.